session.go

```diff
@@ -770,10 +788,22 @@ func (q *Query) execute(conn *Conn) *Iter {
 	return conn.executeQuery(q)
 }
 
-func (q *Query) attempt(d time.Duration) {
+func (q *Query) attempt(end, start time.Time, iter *Iter) {
 	q.attempts++
-	q.totalLatency += d.Nanoseconds()
+	q.totalLatency += end.Sub(start).Nanoseconds()
 	// TODO: track latencies per host and things as well instead of just total
+
+	if q.observer != nil {
+		q.observer.Observe(QueryObservation{
+			ctx:      q.context,
+			keyspace: q.session.pool.keyspace,
+			stmt:     q.stmt,
+			start:    start,
+			end:      end,
+			rows:     iter.numRows - iter.pos,
```

Comment:
iter.pos here should always be 0, is this what you intend?

Reply:
Would it not be 0 in paginated queries?

Reply:
yeah, always 0. Made TestObserve_Pagination to check it